File: rb/features.rb
# ZippopotamusZipCode SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ZippopotamusZipCodeFeatures
  def self.make_feature(name)
    case name
    when "base"
      ZippopotamusZipCodeBaseFeature.new
    when "test"
      ZippopotamusZipCodeTestFeature.new
    else
      ZippopotamusZipCodeBaseFeature.new
    end
  end
end

File: py/features.py
# ZippopotamusZipCode SDK feature factory

from feature.base_feature import ZippopotamusZipCodeBaseFeature
from feature.test_feature import ZippopotamusZipCodeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ZippopotamusZipCodeBaseFeature(),
        "test": lambda: ZippopotamusZipCodeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
